### ISSUE TITLE
[Request] cli command: dotnet add sourcelink

### ISSUE THREAD

**Issue by chuckries** [dotnet/sourcelink#709]
It would be really awesome if the source link package reference could be added by the dotnet cli.

something like 
`dotnet add sourcelink` 
or 
`dotnet add sourcelink github`

**Comment by tmat** [comment]
Would that find and add the latest version?

**Comment by chuckries** [comment]
Ideally, or something else smart.

I can never remember the exact name of the source link packages *AND* the guidance is to throw additional configuration at it (like PriveAssets=all or something). It would be great if the cli could just streamline that process.

Additionally, it would be awesome if we could be more vocal about packages that *don't* have source link enabled. For example, a warning on dotnet pack that yells about a package not having source link.

**Comment by tmat** [comment]
@clairernovotny @richlander

**Comment by tmat** [comment]
Another thing that we could do is to add Source Link reference by default when you create project in VS - VS knows what source control you're using so it can choose the right Source Link package. 

Not sure if that's doable for `dotnet new`. We could implement heuristic looking for the domain of your `origin` remote URL. If it's github.com or visualstudio.com/dev.azure.com then it's obvious. But otherwise we don't know what the host is. We could perhaps provide a list of all supported Git providers and let the user choose (like `Which source control server do you use? [1] GitLab, [2] GitHub Enterprise, ...`).

**Comment by chuckries** [comment]
I like these ideas very much. I still like the idea of warning on `dotnet pack` in scenarios as Source Link is not all that interesting for a throwaway test app, but once you're creating packages you should absolutely have Source Link.

Additionally, for a new project you may run `dotnet new` before running `git init` and there is no SCM info at the time of `dotnet new`

**Comment by tmat** [comment]
Definitely, I agree nuget packaging should report missing source link info.

> I've also always wondered if we could add source link as a build step/build config (in Auzre Pipelines or GH Actions) in order to allow customers to enable it even if the packagereference is not in their project. I think a lot of people would think of Source Link as a build/release time thing, rather than something that exists in the source.

I don't think this is a good approach. The CI should not be tweaking the project build. It is important that CI and local builds are the same (or as close as possible).

**Comment by chuckries** [comment]
@tmat @clairernovotny @richlander It seems like the sourcelink repo is not the correct place for this. Should I file an issue on dotnet cli? NuGet? Roslyn? Where's the best place for this? I'd really like to get traction on this as we are trying to drive adoption and awareness for Source Link.

**Comment by ngardon** [comment]
Where are we on this ?
I'd be interested to try and tackle this, if no one is.

I'd just need an overview of the functionality, and the place to put it, as @chuckries pointed out.

**Comment by chuckries** [comment]
cc @tmat as the thinking around this has changed.
